sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -208,18 +226,37 @@ def send_batch(self, event_data_batch, **kwargs):
                 :caption: Sends event data
 
         """
+        partition_id = kwargs.get("partition_id")
+        partition_key = kwargs.get("partition_key")
+        if isinstance(event_data_batch, EventDataBatch):
+            if partition_id or partition_key:
+                raise TypeError("partition_id and partition_key should be None when sending an EventDataBatch "
+                                "because type EventDataBatch itself may have partition_id or partition_key")
+            to_send_batch = event_data_batch
+        elif isinstance(event_data_batch, List):
+            to_send_batch = self.create_batch(partition_id=partition_id, partition_key=partition_key)
+            for event_data in event_data_batch:
+                try:
+                    to_send_batch.add(event_data)
+                except ValueError:
+                    raise ValueError("The list of EventData exceeds the Event Hub frame size limit. "
+                                     "Please send a smaller list of EventData, or use EventDataBatch, "
+                                     "which is guaranteed to be under the frame size limit")
```

Comment:
I'm thinking of making this part a private method of `EventDataBatch` object to make it reusable and easy to maintain.
```python
class EventDataBatch:
    def _from_list(self, event_data_list):
        try:
            for event in event_data_list:
                self.add(event)
        except ValueError:
            raise ValueError("The list of EventData exceeds the Event Hub frame size limit. "
                                     "Please send a smaller list of EventData, or use EventDataBatch, "
                                     "which is guaranteed to be under the frame size limit")
```

then in sender you just need to:

```python
to_send_batch = self.create_batch(partition_id=partition_id, partition_key=partition_key)
to_send_batch._from_list(event_data_batch)
```

Reply:
Added EventDataBatch._load_events